raster_array/tests/test_raster_array.py

```diff
@@ -1,41 +1,52 @@
 # type: ignore
 """Tests for the RasterArray class."""
 
+from contextlib import contextmanager
 import numpy as np
 import pytest
 import rasterio as rio
 
 
-from raster_array.exceptions import RasterArrayShapeError
+from raster_array.exceptions import RasterArrayShapeError, RasterArrayDtypeError
 from raster_array.raster_metadata import RasterMetadata
-from raster_array.raster_array import RasterArray
+from raster_array.raster_array import RasterArray, ensure_valid_nodata
 
 
-@pytest.fixture(scope="session")
-def raster_4_x_4_multiband():
-    shape = (2, 4, 4)
-    count, height, width = shape
-    n = count * height * width
-    array = np.arange(0, n).reshape(shape)
+@contextmanager
+def generate_raster(data, nodata, dtype):
```

Comment:
This is a generic helper to construct rasters for tests. I really like returning (yielding) a dataset (`rasterio.io.DatasetReader`) rather than an actual raster array or temp file. This allows us to treat the result the same as a result from `with rasterio.open(...) as src`.